testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java

```diff
@@ -2958,4 +2972,69 @@ public void testCreateUserDoNotGrantRole() {
             testRealm().roles().get("realm-role").remove();
         }
     }
+
+    /**
+     * Test for #9482
+     */
+    @Test
+    public void joinParentGroupAfterSubGroup() {
+        String username = "user-with-sub-and-parent-group";
+        String parentGroupName = "parent-group";
+        String subGroupName = "sub-group";
+
+        GroupRepresentation parentGroup = createGroup(realm, GroupBuilder.create().name(parentGroupName).build());
+        GroupRepresentation subGroup = createSubGroup(realm, parentGroup.getId(), GroupBuilder.create().name(subGroupName).build());
+
+        UserRepresentation build = UserBuilder.create()
+                .username(username)
+                .build();
+        String userId = createUser(build);
```

Comment:
Could you leverage `Creator.create(RealmResource, GroupRepresentation)` and `Creator.create(RealmResource, UserRepresentation)` instead in both tests to clean after the test completes?

Example:
https://github.com/keycloak/keycloak/blob/fd5a423b551b9836bda44d47b38d7cd34f8d27c4/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java#L2883-L2894

Reply:
@hmlnarik Since @leischt is on leave, I will step in. I adapted the tests as you requested.

Reply:
Thank you.